### PR TITLE
Fixes camera flickering with Gear controller

### PR DESCRIPTION
**Pull request by NolaDonato**
This patch was submitted by SteveGreatApe as a fix for issue #1686 

GearVRF DCO signed off by: Nola Donato nola.donato@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
Please remove the ResonanceAudio extension from this pull request

**Comment by NolaDonato** [comment]
I have rebased this with master and tested it with the controller apps. I no longer see any crashes.

**Comment by NolaDonato** [comment]
Added changes from @SteveGreatApe and rebased against master to get the API changes. Ready to go now.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRGearCursorController.java [hunk 1]
```diff
@@ -738,43 +659,48 @@ final void recycle() {
...
-    private final ConcurrentLinkedQueue<KeyEvent> mKeyEvents = new ConcurrentLinkedQueue<>();
-    private final ConcurrentLinkedQueue<MotionEvent> mMotionEvents = new ConcurrentLinkedQueue<>();
-    private final Activity mActivity;
 
-    SendEvents(final Activity activity) {
-        mActivity = activity;
-    }
-
-    public void init(List<KeyEvent> keyEvents, List<MotionEvent> motionEvents)
+    public static final class SendEvents implements Runnable
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRGearCursorController.java]: Doesn't have to be public.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRGearCursorController.java [hunk 2]
```diff
@@ -304,41 +298,22 @@ public void setScene(GVRScene scene) {
...
-
     public void onDrawFrame()
     {
         boolean wasConnected = mConnected;
 
         mConnected = (mControllerReader != null) && mControllerReader.isConnected();
         if (!mConnected)
         {
-            if (initialized)
+            if (wasConnected)
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRGearCursorController.java]: I don't think after merging with master the changes to this method do what is intended. There is no gaze cursor controller for me in any app.

**Comment by liaxim** [comment]
It would be nice if the pull request was reduced to the commit with the actual changes. There is too much unrelated stuff that was already merged. Makes it hard to see what is going on. I end up merge-squashing this onto my upstream master and diff-ing there..

**Comment by NolaDonato** [comment]
replaced by #1759

**Pull request closed without merging**